src/pages/javascript/index.md

```diff
@@ -1,10 +1,67 @@
 ---
-title: JavaScript
+title: Data Structure, Queue in ES6
 ---
-## JavaScript
+## What is a Queue?
+>A Queue is a particular kind of abstract data type or collection in which the entities in the collection are kept in order and >the principal operations are the addition of entities to the rear terminal position, known as enqueue, and removal of entities >from the front terminal position, known as dequeue. This makes the Queue a First-In-First-Out (FIFO) data structure. In a FIFO >data structure, the first element added to the Queue will be the first one to be removed. Source [Wikipedia]
+>(https://en.wikipedia.org/wiki/Queue_(abstract_data_type)).
 
-JavaScript is the most widely used scripting language on Earth. And it has the largest library ecosystem of any programming language.
+# Complexity of Queue
+| Access        | Search        | Insertion  | Deletion |
+| ------------- |:-------------:| :---------:|---------:|
+| (O)n          |       (O)n    | (O)1       |  (O)n    |
 
-JavaScript is the core language of the web, and the only programming language that can run in all major web browsers.
 
-If you want to learn more about the JavaScript language, and why it's so widely used, read Quincy Larson's <a href='https://medium.freecodecamp.org/what-programming-language-should-i-learn-first-%CA%87d%C4%B1%C9%B9%C9%94s%C9%90%CA%8C%C9%90%C9%BE-%C9%B9%C7%9D%CA%8Dsu%C9%90-19a33b0a467d' target='_blank' rel='nofollow'>"Which programming language should I learn first?"</a>
\ No newline at end of file
+# The Code
+
+```javascript
+class Queue { // first we create the class Queue
+  constructor() {
+    this.queue = []; // once the class is invoked the queue array will be created
```

Comment:
Instead of using arrays, you can perhaps use an object with 2 pointer or a linked list.
A simple example:
```
class Queue {
  constructor(){
    this.queue = {};
    this.start = 0;
    this.end = 0;
  }
  enqueue(val){ // O(1)
	return this.queue[this.end++] = val;
  }
  dequeue(){ // O(1)
    if (!this.length()) return false;
	return delete this.queue[this.start++];
  }
  length(){
	return this.end - this.start;
  }
}
```
Other methods can be implemented in this class.
Note: The accessibility of objects in JS is of time complexity `O(1)`